pykafka/rdkafka/consumer.py

```diff
@@ -20,18 +21,47 @@
 
 class SimpleConsumer(base.BaseSimpleConsumer):
 
-    def __init__(self, client, topic, partitions=None):
-        if isinstance(topic, basestring):
-            topic = client[topic]
-        self._topic = topic
-        self._partitions = partitions or self.topic.partitions # ie all
+    DEFAULTS = get_defaults_dict(base.BaseSimpleConsumer.__init__)
+
+    def __init__(
+            self,
+            topic,
+            cluster,
+            consumer_group=DEFAULTS["consumer_group"],
+            partitions=None,
+            fetch_message_max_bytes=DEFAULTS["fetch_message_max_bytes"],
+            num_consumer_fetchers=DEFAULTS["num_consumer_fetchers"],
+            auto_commit_enable=DEFAULTS["auto_commit_enable"],
+            auto_commit_interval_ms=DEFAULTS["auto_commit_interval_ms"],
+            queued_max_messages=DEFAULTS["queued_max_messages"],
+            fetch_min_bytes=DEFAULTS["fetch_min_bytes"],
+            fetch_wait_max_ms=DEFAULTS["fetch_wait_max_ms"],
+            refresh_leader_backoff_ms=DEFAULTS["refresh_leader_backoff_ms"],
+            offsets_channel_backoff_ms=DEFAULTS["offsets_channel_backoff_ms"],
+            offsets_commit_max_retries=DEFAULTS["offsets_commit_max_retries"],
+            auto_offset_reset=DEFAULTS["auto_offset_reset"],
+            consumer_timeout_ms=DEFAULTS["consumer_timeout_ms"]):
```

Comment:
So `DEFAULTS`, through `get_defaults_dict()` pulls the values from `base.py` - I was thinking I wouldn't have to remember updating this if we update `base.py` then. But happy to drop that if you prefer!

(Btw, given that `base.py` doesn't enforce anything now that it's not an abc, perhaps this might as well pull defaults straight from `pykafka.simpleconsumer` instead...)